tests/cpptests/test_cpp_trie.cpp

```diff
@@ -121,3 +123,60 @@ TEST_F(TrieTest, testDeepEntry) {
   ASSERT_EQ(maxbuf, ret.size());
   TrieType_Free(t);
 }
+
+/**
+ * This test ensures payload isn't corrupted when the trie changes.
+ */
+TEST_F(TrieTest, testPayload) {
+  char buf1[] = "world";
+  
+  Trie *t = NewTrie(NULL);
```

Comment:
Can we add a test that uses the free callback?

Reply:
Done